test/prepare.rb

```diff
@@ -3,13 +3,31 @@
 require "find"
 require "fileutils"
 
-Find.find("sql") do |entry|
-  if File.directory?(entry)
-    results_directory = entry.gsub(/\Asql/, "results")
-    FileUtils.mkdir_p(results_directory)
-  elsif File.file?(entry)
-    next unless entry.end_with?(".sql")
-    test_file = entry.gsub(/\Asql\/|\.sql\z/, "")
-    puts("test: #{test_file}")
+def open_output(output_path)
+  if output_path
+    File.open(output_path, "w") do |output|
+      yield(output)
+    end
+  else
+    yield($stdout)
+  end
+end
+
+open_output(ARGV[0]) do |output|
+  source_dir = File.dirname(File.dirname(__FILE__))
+  sql_dir = File.join(source_dir, "sql")
+  Find.find(sql_dir) do |entry|
+    if File.directory?(entry)
+      results_directory = entry.gsub(/\A#{sql_dir}/, "results")
```

Comment:
We should escape by `Regexp.escape`:

```suggestion
      results_directory = entry.gsub(/\A#{Regexp.escape(sql_dir)}/, "results")
```

See also: https://www.clear-code.com/blog/2015/3/25.html

Using `Pathname#relative_path_from` may be better.

Reply:
fix: [3d63062](https://github.com/pgroonga/pgroonga/pull/753/commits/3d630623c8b83faa56e3502fe5bf9eba414da29f) Thanks, the code gets much simpler now.